python/mdvtools/dbutils/mdv_server_app.py

```
@@ -810,6 +1324,357 @@ def change_project_access(project_id):
+        def share_project(project_id):
+            """Fetch users with whom the project is shared along with their permissions."""
+            try:
+                print(f"Sharing project '{project_id}'")
+
+                # Step 1: Skip authentication if not enabled
+                if not ENABLE_AUTH:
+                    print("Authentication is disabled, skipping authentication check.")
+                    return jsonify({"message": "Authentication is disabled, no action taken."})
+
+                # Step 2: Check if user is already authenticated in the cache
+                user_id = redis_client.get('user_id')  # Retrieve cached user_id if available
```

Ruff (0.8.2)

Undefined name `redis_client`

(F821)